specs/Print.md

```diff
@@ -0,0 +1,816 @@
+void ShowPrintUI(object target, ExecutedRoutedEventArgs e)
+{
+  string printDialog = e.Parameter.ToString();
+  if (printDialog == "Browser")
+  {
+    // Opens the browser print preview dialog.
+    webView.CoreWebView2.ShowPrintUI();
+  }
+  else
+  {
+    // Opens the system print dialog.
+    webView.CoreWebView2.ShowPrintUI(CoreWebView2PrintDialogKind.System);
+  }
+}
+```
+
+# Print to default printer
+
+You can use `Print` method to print to a default printer without a print dialog.
+You can either use default page and printer settings or programmatically configure
+print settings, and call `Print` to print the current web page.
+
+```cpp
+// This example prints the current web page without a print dialog to a default printer
+// with the default print settings.
+
+bool AppWindow::PrintToDefaultPrinter()
+{
+  wil::com_ptr<ICoreWebView2_15> webView2_15;
+  CHECK_FAILURE(m_webView->QueryInterface(IID_PPV_ARGS(&webView2_15)));
+
+  wil::com_ptr<ICoreWebView2Environment6> webviewEnvironment6;
+  CHECK_FAILURE(m_webViewEnvironment->QueryInterface(IID_PPV_ARGS(&webviewEnvironment6)));
+
+  wil::com_ptr<ICoreWebView2PrintSettings> printSettings = nullptr;
+  CHECK_FAILURE(webviewEnvironment6->CreatePrintSettings(&printSettings));
+
+  wil::com_ptr<ICoreWebView2PrintSettings2> printSettings2;
+  CHECK_FAILURE(printSettings->QueryInterface(IID_PPV_ARGS(&printSettings2)));
+
+  wil::unique_cotaskmem_string title;
+  CHECK_FAILURE(m_webView->get_DocumentTitle(&title));
+
+  // Prints current web page with the default page and printer settings.
+  CHECK_FAILURE(webView2_15->Print(
+      printSettings2.get(),
+      Callback<ICoreWebView2PrintCompletedHandler>(
+          [title = title.get(), this](HRESULT errorCode, BOOL isSuccessful) -> HRESULT
+          {
+            std::wstring message = L"";
+            if (errorCode == S_OK && isSuccessful)
+            {
+              message = L"Printing " + std::wstring(title) +
+                              L" document to printer is succedded";
+            }
+            else
+            {
+              if (errorCode == S_OK)
+              {
+                message = L"Printer is not available, offline or error state";
+              }
+              else if (errorCode == E_ABORT)
+              {
+                message = L"Printing " + std::wstring(title) +
+                                  L" document already in progress";
+              }
+              else if (errorCode == E_FAIL)
+              {
+                message = L"Printing " + std::wstring(title) +
+                                  L" document to printer is failed";
+              }
+            }
+
+            AsyncMessageBox(message, L"Print to default printer");
+
+            return S_OK;
+        })
+        .Get()));
+  return true;
+}
+```
+
+```c#
+async void PrintToDefaultPrinter ()
+{
+  try
+  {
+    CoreWebView2PrintSettings printSettings = null;
+    printSettings = WebViewEnvironment.CreatePrintSettings();
```

Comment:
What's the point of using PrintSettings if it's just the default values being used?

Reply:
Yes, it is not necessary. Will remove.

Reply:
Please fix (and document if that's not already covered in the documentation)